apps/web/src/components/Navigation/problem-explorer-track-nav.tsx

```diff
@@ -50,11 +52,23 @@ export function ProblemExplorerTrackNav({ isCollapsed, className }: ProblemExplo
         className,
       )}
     >
+      <Button
```

Comment:
I think we need to put in a bit more thought on the design here for when the panel is collapsed.

Right now, the back button is just pushing the existing button to Explore drawer to the side, which is not ideal.



Before:

https://github.com/user-attachments/assets/02d21ec6-f54b-420a-a8ec-19b2a2e8f901


After:
https://github.com/user-attachments/assets/fb198819-ab24-4425-9634-b3d32d650d43

Reply:
How about something like this?
<img width="332" alt="image" src="https://github.com/user-attachments/assets/1e2734e3-de93-4dfc-887f-cdcf85d7cea8">

Reply:
That's definitely an improvement! I'll defer to the others to chime in here, also @PickleNik is really good with UI!

My only nit here is that I think it needs separation with a divider similar to the section below, but then you have two sections with just one icon. Not sure.

Reply:
Would it make sense to conditionally render it if you can go back to typehero?